protocols/api/src/main/java/org/apache/james/protocols/api/OIDCSASLParser.java

```diff
@@ -61,11 +69,25 @@ public static Optional<OIDCInitialResponse> parse(String initialResponse) {
             String decodeValueWithoutDanglingPart = decodeResult.filter(value -> value.startsWith("n,"))
                 .map(value -> value.substring(2))
                 .orElse(decodeResult.get());
-            List<String> parts = Splitter.on(SASL_SEPARATOR).splitToList(decodeValueWithoutDanglingPart);
-            ImmutableList<String> tokenPart = parts.stream().filter(part -> part.startsWith("auth=")).collect(ImmutableList.toImmutableList());
-            ImmutableList<String> userPart = parts.stream().filter(part -> part.startsWith("user=")).collect(ImmutableList.toImmutableList());
-            if (tokenPart.size() == 1 && userPart.size() == 1) {
-                return Optional.of(new OIDCInitialResponse(userPart.get(0).substring(5), tokenPart.get(0).substring(5).replace(PREFIX_TOKEN, "")));
+
+            StringTokenizer stringTokenizer = new StringTokenizer(decodeValueWithoutDanglingPart, String.valueOf(SASL_SEPARATOR));
+            String tokenPart = null;
+            String userPart = null;
+            String associatedUserPart = null;
+
+            while (stringTokenizer.hasMoreTokens()) {
+                String stringToken = stringTokenizer.nextToken();
+                if (stringToken.startsWith("auth=")) {
+                    tokenPart = stringToken.substring(5).replace(PREFIX_TOKEN, "");
```

Comment:
Arg we should not use `String::replace` as thisis backed by regexps.

Have a look at StringUtils from commons-lang 3 library